store/daystore/day_store.go

```diff
@@ -79,7 +79,8 @@ func NewDayStore(epsilon uint64, resolution time.Duration) *DayStore {
 	// Calculate how many resolutions correspond to an hour
 	hourNS := time.Hour.Nanoseconds()
 	resNS := resolution.Nanoseconds()
-	intervals := uint8(hourNS / resNS)
+	intervals := uint16(hourNS / resNS)
```

Comment:
As a side note - if your goal is to optimize memory at all costs, what you can do to avoid padding is to use slices or arrays instead of multiple fields of different sizes. But I don't think we need that level of premature optimization - at least not until it is proved necessary.